flang/include/flang/Optimizer/Builder/Runtime/RTBuilder.h

```diff
@@ -776,6 +776,9 @@ static mlir::func::FuncOp getRuntimeFunc(mlir::Location loc,
   if (func)
     return func;
   auto funTy = RuntimeEntry::getTypeModel()(builder.getContext());
+  if (funTy.getResults().size() == 1 &&
+      mlir::isa<mlir::NoneType>(funTy.getResult(0)))
+    funTy = mlir::FunctionType::get(funTy.getContext(), funTy.getInputs(), {});
```

Comment:
Is it possible to move this handling in [getTypeModel](https://github.com/llvm/llvm-project/blob/3eee79ae71d9ce800c760c652aab5bf90e1d78d1/flang/include/flang/Optimizer/Builder/Runtime/RTBuilder.h#L677) here directly so that any other current/future usage of  `getTypeModel` are correct?

Reply:
Yes that looks like a better place. I'll update the patch